### PR TITLE
CLDR-7428 Freeze collators; finishing touches

### PR DESCRIPTION
**Pull request by btangmu**
-Remove TODO comment in Misc.java, add comment explaining why no freeze

-Use ROOT_COLLATOR in CLDRModify.java

-Use ROOT_SECONDARY in ExtractMessages.java

CLDR-7428

- [x] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: tools/cldr-code/src/main/java/org/unicode/cldr/draft/Misc.java [hunk 1]
```diff
@@ -144,12 +144,11 @@ private static void showDefaultContent(String... strings) {
 
     private static void showSortKey() {
         String[] tests = "a ä A ぁ あ ァ ｧ ア ｱ ㋐".split(" ");
-        // TODO: freeze the Collator; problematic since changed in innermost for loop below
-        // Reference: https://unicode-org.atlassian.net/browse/CLDR-7428
         RuleBasedCollator c = (RuleBasedCollator) Collator.getInstance(ULocale.ROOT);
         c.setStrength(RuleBasedCollator.QUATERNARY);
         c.setCaseLevel(true);
         c.setHiraganaQuaternary(true);
+        // Do not freeze the collator since it is changed in the innermost "for" loop below
```
Comment (btangmu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/draft/Misc.java:151]: this method is only called by main() in this file

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ExtractMessages.java [hunk 2]
```diff
@@ -205,12 +206,7 @@ public void handlePathValue(String path, String value) {
         }
     }
 
-    public static Collator col = Collator.getInstance(ULocale.ROOT); // freeze below
-
-    static {
-        col.setStrength(Collator.SECONDARY);
-        col = col.freeze();
-    }
+    public static Collator col = CollatorHelper.ROOT_SECONDARY;
```
Comment (btangmu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ExtractMessages.java:209]: Mark, you pointed out this one when reviewing the previous PR

File: tools/cldr-code/src/main/java/org/unicode/cldr/draft/Misc.java (hunk 1 above)
Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/draft/Misc.java:151]: ok

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ExtractMessages.java (hunk 2 above)
Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/tool/ExtractMessages.java:209]: ah

**Review by macchiati — APPROVED** [review]

**Pull request merged**